sdk/appcenter/src/main/java/com/microsoft/appcenter/utils/context/AuthTokenContext.java

```diff
@@ -352,16 +358,25 @@ public synchronized void removeOldestTokenIfMatching(String token) {
      * @param authTokenInfo auth token to check for expiration.
      */
     public void checkIfTokenNeedsToBeRefreshed(@NonNull AuthTokenInfo authTokenInfo) {
+        RefreshListener refreshListener = mRefreshListener.get();
+        if (refreshListener != null) {
+            String accountId = getAccountIdToRefreshAndUpdateCache(authTokenInfo);
+            if (accountId != null) {
+                refreshListener.onTokenRequiresRefresh(accountId);
+            }
+        }
+    }
+
+    private synchronized String getAccountIdToRefreshAndUpdateCache(@NonNull AuthTokenInfo authTokenInfo) {
```

Comment:
```suggestion
    /**
     * Check if the last/current token needs a refresh because expiring soon.
     *
     * @param authTokenInfo Auth token to check.
     * @return Home accountId with last token about to expire or already expired.
     */
    private synchronized String getHomeAccountIdToRefreshToken(@NonNull AuthTokenInfo authTokenInfo) {
```

Reply:
@guperrot's suggestion. Took it.